src/app/components/SearchForm/SearchForm.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import SearchButton from '../Button/SearchButton';
+
+class SearchForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.allowedFields = props;
+
+    this.onFieldChange = this.onFieldChange.bind(this);
+    this.onQueryChange = this.onQueryChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.submitSearchRequest = this.submitSearchRequest.bind(this);
+  }
+
+  onFieldChange(event) {
+    const fieldSelected = event.target.value;
+    this.setState({ searchField: fieldSelected });
+  }
+
+  onQueryChange(event) {
+    this.setState({ searchQuery: event.target.value });
+  }
+
+  handleSubmit(event) {
+    if (event && event.charCode === 13) {
+      this.submitSearchRequest(event);
+    }
+  }
+
+  submitSearchRequest(event) {
+    event.preventDefault();
+    if (!this.state.searchField) {
+      this.setState({ searchField: 'q' });
+    }
+
+    const terms = this.state.searchQuery.trim().replace(/\s+/g, ' ');
+    this.props.search(terms, this.state.searchField);
+  }
+
+  render() {
+    return (
+      <div className="nypl-row">
+        <div className="nypl-column-full">
+          <form className="nypl-omnisearch-form" onSubmit={this.handleSubmit} onKeyPress={this.handleSubmit}>
+            <div className="ebook-search-form">
+              <div className="nypl-omnisearch">
+                <div className="nypl-text-field">
+                  <span className="nypl-omni-fields">
+                    <label htmlFor="search-by-field">Search in</label>
+                    <select
+                      id="search-by-field"
+                      onChange={this.props.onFieldChange}
+                    >
+                      <option value={this.props.allowedFields.kw} defaultValue>
+                        Keyword
+                      </option>
+                      <option value={this.props.allowedFields.ti}>
+                        Title
+                      </option>
+                      <option value={this.props.allowedFields.au}>
+                        Author
+                      </option>
+                    </select>
+                  </span>
+                </div>
+                <div className="nypl-text-field">
+                  <span className="nypl-omni-fields-text">
+                    {/* <label htmlFor="search-input">Search ResearchNow for</label> */}
```

Comment:
Oh, I think you do need this label element eventually.

Reply:
Yeah -- I commented out because it was causing display issues, but left it in because it's needed.